### PR TITLE
File_Q&A Default App + Switch from Anthropic to OpenRouter

### PR DESCRIPTION
**Pull request by NolanGC**
## Related Linear Issues

[OPE-81](https://linear.app/openrouter/issue/OPE-81/remove-anthropic-and-replace-with-openai) - remove anthropic and replace with openai
[OPE-85](https://linear.app/openrouter/issue/OPE-85/thinking-we-should-make-file-qanda-the-default-and-have-basic-chat-be) - make file q&a the default

## Changes Made

- Made File_Q&A the default app, moved Chatbot into the pages directory
- Removed Anthropic from File Q&A and replaced with OpenRouter OpenAI calls
- Workaround fix for recurring error with openai.ChatCompletion.create sometimes returning a string

## Additional Notes
Conditional `json.loads` is a hacky fix but eliminates the error with openai.ChatCompletion completely. `#TODO` left behind to revisit in the future.
```python
if(type(response) == str):
       response = json.loads(response)
  msg = response["choices"][0]["message"]
```

### CODE REVIEW

File: File_Q&A.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # openai.api_base = "http://localhost:3000/api/v1"
+    openai.api_key = api_key
+    context_prompt = f"""Here's an article:\n\n<article>
+    {article}\n\n</article>\n\n"""
+    context_message = {"role": "assistant", "content": context_prompt}
+    question_message = {"role": "user", "content": question}
+    response = openai.ChatCompletion.create(
+        model="openai/gpt-3.5-turbo",
+        messages=[context_message, question_message],
+        headers={"HTTP-Referer": "https://yourdomain.streamlit.io"},
```
Comment (alexanderatallah) [inline File_Q&A.py:34]: let's move this referrer value out into a constant? so we can set it everywhere

File: File_Q&A.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    )
+    # response is sometimes type str
```
Comment (alexanderatallah) [inline File_Q&A.py:36]: think we should figure out the root cause of this

File: File_Q&A.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    article = uploaded_file.read().decode()
+    openai.api_base = "https://openrouter.ai/api/v1"
+    # openai.api_base = "http://localhost:3000/api/v1"
+    openai.api_key = api_key
+    context_prompt = f"""Here's an article:\n\n<article>
+    {article}\n\n</article>\n\n"""
+    context_message = {"role": "assistant", "content": context_prompt}
+    question_message = {"role": "user", "content": question}
+    response = openai.ChatCompletion.create(
+        model="openai/gpt-3.5-turbo",
```
Comment (alexanderatallah) [inline File_Q&A.py:32]: we should probably default to claude 100k (though the model will be selectable in the sidebar)

File: File_Q&A.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        disabled=not uploaded_file,
+    )
+    b.form_submit_button("Ask", use_container_width=True)
+
+if uploaded_file and question and not api_key:
+    st.info("Please add your OpenRouter API key to continue.")
+
+if uploaded_file and question and api_key:
+    article = uploaded_file.read().decode()
+    openai.api_base = "https://openrouter.ai/api/v1"
```
Comment (alexanderatallah) [inline File_Q&A.py:24]: let's move this to a constant.

**Review by alexanderatallah — CHANGES_REQUESTED** [review]
Should we close this PR in favor of #2 , since it seems to build on top of it and merge into main?

**Comment by NolanGC** [comment]
Sure we can work on merging #2 and close this one. What to do about the requested changes here?

**Pull request closed without merging**